source/StarField.cpp

```diff
@@ -153,12 +171,30 @@ void StarField::Draw(const Point &pos, const Point &vel, double zoom) const
 	DrawList drawList;
 	drawList.Clear(0, zoom);
 	drawList.SetCenter(pos);
+
+	if(hazeCloak > 0.)
+		hazeCloak -= FADE_PER_FRAME;
+	else
+	{
+		hazeCloak = 0.;
+		isFading = false;
+	}
 	
 	// Any object within this range must be drawn. Some haze sprites may repeat
 	// more than once if the view covers a very large area.
 	Point size = Point(1., 1.) * haze.front().Radius();
 	Point topLeft = pos + (Screen::TopLeft() - size) / zoom;
 	Point bottomRight = pos + (Screen::BottomRight() + size) / zoom;
+	DrawHaze(haze, topLeft, bottomRight, hazeCloak, drawList);
+	if(hazeCloak > 0.)
+		DrawHaze(prevHaze, topLeft, bottomRight, 1 - hazeCloak, drawList);
+
+	drawList.Draw();
```

Comment:
I tried to use BatchDrawList but I am sorry that I was not able to do it. I am still studying how BatchDrawList actually works.

Reply:
Yes, that's fine. I briefly looked at it and the batch shader is not configured to support sprite transparency / cloaking.

Given the very limited amount of time haze fades in and out relative to the time haze is just being drawn at full opacity, a future PR could certainly optimize the base case to use the batch shader, or to use the engine draw stack's existing list rather than a new one.